python/ml_croissant/ml_croissant/_src/structure_graph/nodes/source.py

```diff
@@ -101,3 +102,25 @@ def from_json_ld(cls, issues: Issues, field: Any) -> "Source":
     def __bool__(self):
         """Allows to write `if not node.source` / `if node.source`"""
         return len(self.reference) > 0
+
+
+def _apply_transform_fn(value: str, transform: Transform) -> str:
+    if transform.regex is not None:
+        source_regex = re.compile(transform.regex)
+        match = source_regex.match(value)
+        if match is None:
+            logging.debug(f"Could not match {source_regex} in {value}")
+            return value
+        for group in match.groups():
+            if group is not None:
+                return group
+    return value
+
+
+def apply_transforms_fn(value: str, source: Source | None = None) -> str:
```

Comment:
As a general rule, I think it would be good to have a one-line docstring for all functions (even the ones for which the function name is somewhat already "explicative") :)

Reply:
Done.